packages/service-library/src/servicelib/archiving_utils.py

```diff
@@ -141,49 +161,81 @@ async def unarchive_dir(
                 for zip_entry in zip_file_handler.infolist()
             ]
 
-            extracted_paths: Tuple[Path] = await asyncio.gather(*tasks)
+            extracted_paths: list[Path] = await tqdm_asyncio.gather(
```

Comment:
So here the progress is reported as files finished being decompressed, right?

Reply:
yes, because as it is using a process pool I cannot pass the progress bar class since it needs to be pickle compatible.
therefore you get several sub-progress bars with the file name